### ISSUE TITLE
Issue: XGBoost モデル実装

### ISSUE THREAD

**Issue by syunyafukuda** [syunyafukuda/Hull-Tactical---Market-Prediction#85]
## 概要

モデル選定フェーズにおいて、XGBoostモデルの学習スクリプトを実装する。
LGBMベースラインとの比較およびアンサンブル用の多様性確保が目的。

## 背景

- **現在のベストスコア**: LB 0.681（LGBM, FS_compact 116列）
- **OOF RMSE**: 0.012164（LGBM ベースライン）
- **期待効果**: LGBM との予測相関が適度に異なり（< 0.98）、アンサンブル効果が見込める

### 必須タスク

- [ ] `src/models/xgboost/train_xgb.py` - メイン学習スクリプト
- [ ] `configs/models/xgboost.yaml` - YAML設定ファイル
- [ ] `tests/models/test_xgboost.py` - ユニットテスト

### 成果物

- [ ] `artifacts/models/xgboost/inference_bundle.pkl`
- [ ] `artifacts/models/xgboost/oof_predictions.csv`
- [ ] `artifacts/models/xgboost/cv_fold_logs.csv`
- [ ] `artifacts/models/xgboost/model_meta.json`

### パイプライン構造

```
data/raw/train.csv
    ↓
[SU1 + SU5 特徴量生成]  # 既存モジュール再利用
    ↓
577列
    ↓
[tier3 feature exclusion]
    ↓
116列 (FS_compact)
    ↓
[SimpleImputer + 前処理]
    ↓
[XGBRegressor]
```

### 初期ハイパーパラメータ

```python
xgb_params = {
    "objective": "reg:squarederror",
    "tree_method": "hist",
    "max_depth": 6,
    "learning_rate": 0.05,
    "n_estimators": 600,
    "subsample": 0.8,
    "colsample_bytree": 0.8,
    "reg_alpha": 0.0,
    "reg_lambda": 1.0,
    "min_child_weight": 32,
    "random_state": 42,
    "n_jobs": -1,
    "verbosity": 0,
}
```

### LGBMとのパラメータ対応

| LGBM | XGBoost | 初期値 |
|------|---------|--------|
| `num_leaves` | N/A（`max_depth`で制御） | - |
| `max_depth` | `max_depth` | 6 |
| `min_data_in_leaf` | `min_child_weight` | 32 |
| `feature_fraction` | `colsample_bytree` | 0.8 |
| `bagging_fraction` | `subsample` | 0.8 |
| `learning_rate` | `learning_rate` | 0.05 |
| `n_estimators` | `n_estimators` | 600 |

### CV設定

- **手法**: TimeSeriesSplit
- **分割数**: 5
- **gap**: 0
- **Early stopping**: 50 rounds

### 1. 既存モジュールの再利用

```python
from src.feature_generation.su5.train_su5 import (
    load_su1_config,
    load_su5_config,
    load_preprocess_policies,
    SU5FeatureAugmenter,
    _prepare_features,
    load_table,
    infer_train_file,
    infer_test_file,
)
from src.models.common.feature_loader import get_excluded_features
from src.models.common.cv_utils import compute_fold_metrics, evaluate_oof_predictions
```

### 2. 参考実装

- [src/models/lgbm/train_lgbm.py](../src/models/lgbm/train_lgbm.py) をベースに、モデル部分のみ置換

### 3. argparse 引数

```python
parser.add_argument("--data-dir", type=str, default="data/raw")
parser.add_argument("--out-dir", type=str, default="artifacts/models/xgboost")
parser.add_argument("--config-path", type=str, default="configs/feature_generation.yaml")
parser.add_argument("--preprocess-config", type=str, default="configs/preprocess.yaml")
parser.add_argument("--feature-tier", type=str, default="tier3")
parser.add_argument("--n-splits", type=int, default=5)
parser.add_argument("--gap", type=int, default=0)
# XGBoost ハイパラ
parser.add_argument("--max-depth", type=int, default=6)
parser.add_argument("--learning-rate", type=float, default=0.05)
parser.add_argument("--n-estimators", type=int, default=600)
parser.add_argument("--subsample", type=float, default=0.8)
parser.add_argument("--colsample-bytree", type=float, default=0.8)
```

### 4. Early Stopping の実装

```python
model.fit(
    X_train, y_train,
    eval_set=[(X_val, y_val)],
    early_stopping_rounds=50,
    verbose=False,
)
```

### 5. 特徴量名のサニタイズ

XGBoostは特徴量名に `[`, `]`, `<` を含むと警告が出るため、必要に応じてサニタイズ：

```python
import re
feature_cols = [re.sub(r'[\[\]<>]', '_', col) for col in feature_cols]
```

## 成功基準

| 指標 | 閾値 | 理由 |
|------|------|------|
| OOF RMSE | ≤ 0.0125 | ベースライン（0.01216）+3%以内 |
| 予測相関（vs LGBM） | < 0.98 | アンサンブル効果の確保 |
| 品質チェック | pass | ruff, pyright |
| テスト | pass | pytest |

### 学習実行

```bash
# デフォルト設定
uv run python -m src.models.xgboost.train_xgb

# カスタム設定
uv run python -m src.models.xgboost.train_xgb \
    --max-depth 8 \
    --learning-rate 0.03 \
    --n-estimators 800
```

### テスト実行

```bash
uv run pytest tests/models/test_xgboost.py -v
```

### 品質チェック

```bash
uv run ruff check src/models/xgboost/
uv run ruff format src/models/xgboost/
uv run pyright src/models/xgboost/
```

## 依存パッケージ

```bash
uv add xgboost
```

## 参照ドキュメント

- [XGBoost仕様書](../docs/models/xgboost.md) - 詳細な技術仕様
- [LGBM実装](../src/models/lgbm/train_lgbm.py) - 参考実装
- [CV共通モジュール](../src/models/common/cv_utils.py) - CV utilities
- [Feature Loader](../src/models/common/feature_loader.py) - Feature exclusion utilities

## 注意事項

1. **Early Stopping**: `eval_set` の指定が必須
2. **メモリ**: `tree_method="hist"` でGPUなしでも高速化可能
3. **再現性**: マルチスレッド環境では完全な再現性が保証されない場合がある
4. **出力フォーマット**: LGBM と同一フォーマットで出力すること

## ブランチ戦略
切り出し元・マージ先：feat/model-xgboost

---

**Labels**: `model-selection`, `xgboost`, `enhancement`

**Milestone**: Model Selection Phase

**Assignee**: (assign to engineer)
